monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -551,6 +550,7 @@ def get_report(cls):
             raise RuntimeError("Agent repository does not exists")
 
         if cls.report_is_outdated():
-            cls._report = safe_generate_regular_report()
+            with cls._report_generation_lock:
```

Comment:
UI will be periodically sending requests to the Island and all of them will hang here, waiting for report generation lock. Previously if report wasn't generated yet it would return `False` and not hang here

Reply:
The report is generated when we click on security report right? How is the UI sending request to generate the report if we don't click??

Reply:
It's sent every 10 seconds by the looks of it: 
![image](https://user-images.githubusercontent.com/36815064/203530376-35fbc395-0e3a-4043-b68f-0a0d33a30fc0.png)